### ISSUE TITLE
Houston we have a motherfucking TIFF problem

### ISSUE THREAD

**Issue by fish2000** [fish2000/libimread#3]
Observe:
![screen shot 2016-07-29 at 6 49 01 am](https://cloud.githubusercontent.com/assets/378969/17246229/b27e42e2-5558-11e6-88be-11878ca889c5.jpg)
_Figure 1. TIFF Multiwrite with `im.Batch`._

![screen shot 2016-07-29 at 6 46 22 am](https://cloud.githubusercontent.com/assets/378969/17246243/c5f6ccd6-5558-11e6-9569-975087cfc359.jpg)
_Figure 2. What actually happens: left: singular written TIFF file; right: multiwrite TIFF; center: source material._

OK so:
- Width and height are swapped, in a fucked way?
- Planes are read as interleaved during multiwrite (but NOT singular write)?!
- The TIFF library is doing that boundary-condition-repeat shit visible in the singular-write image; I did not specify that – I mean that is nicer than a segfault I guess, but where does it fucking come from???
- Basically those are my chief concerns dogg

**Comment by fish2000** [comment]
As of [that last commit](https://github.com/fish2000/libimread/commit/97657e87cf2c426084ef43dfe816a1b7a1814d08), shit is not exactly better:

![screen shot 2016-07-29 at 7 42 01 am](https://cloud.githubusercontent.com/assets/378969/17247263/0a720310-5560-11e6-8df8-bf8036243f06.jpg)

… what the fuuuuuuuuuuuuuck

**Comment by fish2000** [comment]
Note how the TIFF image is the only one that is fucked in this peculiar pseudo-interleaved kind of way:

![screen shot 2016-07-29 at 7 48 54 am](https://cloud.githubusercontent.com/assets/378969/17247388/f327f25e-5560-11e6-99d8-aeab4cb70cf0.jpg)

![screen shot 2016-07-29 at 7 47 48 am](https://cloud.githubusercontent.com/assets/378969/17247373/d9126c14-5560-11e6-981a-214dba53ced6.jpg)

![screen shot 2016-07-29 at 7 47 44 am](https://cloud.githubusercontent.com/assets/378969/17247377/dbb3f15e-5560-11e6-9c06-6fab74e22ab4.jpg)

**Comment by fish2000** [comment]
Ahem: http://www.awaresystems.be/imaging/tiff/tifftags/planarconfiguration.html

**Comment by fish2000** [comment]
Ahahaha yeah regarding that last:

![screen shot 2016-07-29 at 8 00 39 am](https://cloud.githubusercontent.com/assets/378969/17247649/a3346c26-5562-11e6-9603-2f146cec6e3d.jpg)

… what you get with `PLANARCONFIG_SEPARATE` is, ah, a step back at best

**Comment by fish2000** [comment]
_Clearly tho I suck at basic arithmetic:_

![screen shot 2016-07-29 at 9 28 11 am](https://cloud.githubusercontent.com/assets/378969/17249645/e670e6ca-556e-11e6-8673-5bd6b7c09089.jpg)

… well OK dogg I do suppose this is at least a step foreward

**Comment by fish2000** [comment]
_At least it looks good, provided I tell you I intentionally designed it to be that way:_

![screen shot 2016-07-29 at 9 54 40 am](https://cloud.githubusercontent.com/assets/378969/17250528/b679460c-5572-11e6-8626-431709bb20e1.jpg)

**Comment by fish2000** [comment]
Original source of that last mangled image –

![img_4332](https://cloud.githubusercontent.com/assets/378969/17250822/1232c6e8-5574-11e6-8107-4f6e4663ff4c.jpg)

_(© 2012 me and mine)_

… which should make it obvious what is happening, and what is failing to happen
